### DISCUSSION TITLE
Is there a way to set the OSX and iOS SDK Version?

### DISCUSSION THREAD

**Discussion by TyelorD** [axmolengine/axmol#1791] [Q&A] [answered]
I assume this is relatively straight forward, and is likely just a command line argument when running either "axmol build" or "axmol new" via PowerShell, but I didn't see it documented anywhere.

When I made a new project it defaulted all the targets to macOS 14.1, since that's what is installed on my machine. But I'd like to target an older version of Mac, this way the game runs on older Macs as well as newer ones.

I manually set all of the targets to OSX 10.15, however several third party libraries are pre-compiled, rather than compiled from source at build time. During linking this gives me numerous warnings about: libcrypto.a, libcurl.a, libjpeg.a, and libssl.a, all being built for a newer macOS than being linked. Specifically libjpeg was built for 12.7, and the other three were built for 12.0.

Do I need to download the source for those libraries myself, and compile them on my machine targeting the appropriate version of macOS that I want? Or is there a way to tell the Axmol build scripts to do this for me?

Thanks for your time.

**Comment by halx99** [comment]
since v81, already build for macos 10.13

**Comment by halx99** [comment, marked as answer]
`axmol build -p osx -minsdk 10.15` should work on axmol git latest.

**Reply by TyelorD** [reply]
I can write this in a separate Discussions post if you wish, but how do I set the name for the build directory?

It defaults to `build_$archs`, but for macOS we are using `-a 'arm64 x84_64'` in order to allow the game to run on both M1 Macs as well as Intel Macs: otherwise it defaults to the build machine architecture (arm64 in this case). This makes the build folder's name very ugly and unclear as "build_arm64 x86_64", so I'd prefer if we could set a flag so that the build folder is named something like "build_osx" instead. Looking at the 1k/build.ps1 it looked like I could do this by setting a flag like `-B "build_osx"`, however this didn't change anything and it still defaulted to using "build_arm64 x86_64".

**Reply by halx99** [reply]
-xc '-Bbuild_dir'

**Reply by TyelorD** [reply]
Hmm, actually just targeting both arm64 and x86_64 causes the build to fail with four undefined symbols from libpng.a[arm64]

I'll go ahead and create a new Discussions post for that, which could be moved to the Issues if needed.

**Reply by halx99** [reply]
osx combine archs is more complex, and we no plan to support it
